src/xunit.v3.core/Framework/FactDiscoverer.cs

```diff
@@ -84,11 +86,15 @@ static ExecutionErrorTestCase ErrorTestCase(
 
 		var details = TestIntrospectionHelper.GetTestCaseDetails(discoveryOptions, testMethod, factAttribute);
 
-		return new(
+		var errorTestCase = new ExecutionErrorTestCase(
 			details.ResolvedTestMethod,
 			details.TestCaseDisplayName,
 			details.UniqueID,
 			message
 		);
+
+		errorTestCase.SourceFilePath = factAttribute.SourceFilePath;
+		errorTestCase.SourceLineNumber = factAttribute.SourceLineNumber;
```

Comment:
It's probably better to have these information returned from `GetTestCaseDetails`, and update `ExecutionErrorTestCase` constructor to accept these.

I just went with something quick as a "Poc".